src/os/bluestore/BlueStore.cc

```diff
@@ -1640,6 +1640,31 @@ int BlueStore::_setup_block_symlink_or_file(
 	  VOID_TEMP_FAILURE_RETRY(::close(fd));
 	  return r;
 	}
+#ifdef HAVE_POSIX_FALLOCATE
+	r = ::posix_fallocate(fd, 0, size);
+	if (r < 0) {
+	  r = -errno;
+	  derr << __func__ << " failed to prefallocate " << name << " file to "
+	       << size << ": " << cpp_strerror(r) << dendl;
+	  VOID_TEMP_FAILURE_RETRY(::close(fd));
+	  return r;
+	}
+#else
+	char data[1024*128];
```

Comment:
Yes, found this bug was in my testing env.  Now i am writing code about PMEMdevice(like NVMEDevice) which can use file instead(for wal of rocksdb). So this isn't only for testing usage.